packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx

```diff
@@ -145,25 +179,39 @@ export const DeclarativeChart: React.FunctionComponent<DeclarativeChartProps> =
               canSelectMultipleLegends: true,
               selectedLegends: activeLegends,
             }}
+            componentRef={chartRef}
           />
         );
       }
       return (
         <VerticalStackedBarChart
           {...transformPlotlyJsonToVSBCProps(plotlySchema, colorMap, isDarkTheme)}
           legendProps={legendProps}
+          componentRef={chartRef}
         />
       );
     case 'heatmap':
-      return <HeatMapChart {...transformPlotlyJsonToHeatmapProps(plotlySchema)} legendProps={legendProps} />;
+      return (
+        <HeatMapChart
+          {...transformPlotlyJsonToHeatmapProps(plotlySchema)}
+          legendProps={legendProps}
+          componentRef={chartRef}
+        />
+      );
     case 'sankey':
-      return <SankeyChart {...transformPlotlyJsonToSankeyProps(plotlySchema, colorMap, isDarkTheme)} />;
+      return (
+        <SankeyChart
+          {...transformPlotlyJsonToSankeyProps(plotlySchema, colorMap, isDarkTheme)}
+          componentRef={chartRef}
+        />
+      );
     case 'indicator':
-      if (data?.[0]?.mode?.includes('gauge')) {
+      if (plotlySchema?.data?.[0]?.mode?.includes('gauge')) {
         return (
           <GaugeChart
             {...transformPlotlyJsonToGaugeProps(plotlySchema, colorMap, isDarkTheme)}
             legendProps={legendProps}
+            componentRef={chartRef}
```

Comment:
>componentRef [](http://example.com/codeflow?start=12&length=12)

can ref create perf overhead. Do they need any destruction

Reply:
The component instance is assigned to the ref only during lifecycle methods, so it should not create a significant performance overhead.